core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -311,12 +317,16 @@ object SparkSubmit extends CommandLineUtils {
     }
 
     // In client mode, download remote files.
+    var localPrimaryResource: String = null
+    var localJars: String = null
+    var localPyFiles: String = null
+    var localFiles: String = null
     if (deployMode == CLIENT) {
```

Comment:
If you want to avoid download for yarn, can we just check the cluster mode it here?